core/src/main/java/bisq/core/btc/model/AddressEntryList.java

```diff
@@ -110,16 +111,38 @@ public void onWalletReady(Wallet wallet) {
             list = new ArrayList<>();
             add(new AddressEntry(wallet.freshReceiveKey(), AddressEntry.Context.ARBITRATOR));
 
-            // In case we restore from seed words and have balance we need to add the relevant addresses to our list:
+            // In case we restore from seed words and have balance we need to add the relevant addresses to our list.
+            // IssuedReceiveAddresses does not contain all addressed where we expect balance so we need to listen to
+            // incoming txs at blockchain sync to add the rest.
             if (wallet.getBalance().isPositive()) {
                 wallet.getIssuedReceiveAddresses().forEach(address -> {
-                    log.info("Create AddressEntry for address={}", address);
+                    log.info("Create AddressEntry for IssuedReceiveAddress. address={}", address.toString());
                     add(new AddressEntry((DeterministicKey) wallet.findKeyFromPubHash(address.getHash160()), AddressEntry.Context.AVAILABLE));
                 });
             }
-
             persist();
         }
+
+        // We add a confidence listener to get notified about potential new transactions and
+        // add an address entry list in case it does not exist yet. This is mainly needed for restore from seed words
+        // but can help as well in case the addressEntry list would miss an address where the wallet was received
+        // funds (e.g. if the user sends funds to an address which has not been provided in the main UI - like from the
+        // wallet details window).
+        wallet.addTransactionConfidenceEventListener((w, tx) -> {
```

Comment:
TransactionConfidenceEventListener does the job but it seems to be an overkill. TransactionConfidenceEventListener is invoked for each wallet tx every time its "confidence" is updated (eg, when included in a block and every time a new block is mined on top of that block for the next 100 blocks).

Probably you want to use WalletCoinsReceivedEventListener and WalletCoinsSentEventListener (and  ScriptsChangeEventListener if you use the watched script feature of Wallet) instead to be notified of new addresses.

Reply:
I tried the WalletCoinsReceivedEventListener and WalletCoinsSentEventListener (both) but the balance was not correct that way. I have not investigated further why, but if you could find out would be good. I am aware that the TransactionConfidenceEventListener is a bit heavy.
If you test it you need a wallet which is a bit older and has a few addresses where it has send/received. During the spv chain sync you can check if the UI balance is the same as the one from the wallet shown at the popup opened from cmd+e.